src/openrct2/actions/ChangeMapSizeAction.h

```diff
@@ -0,0 +1,30 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2021 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "../world/Map.h"
+#include "GameAction.h"
+
+DEFINE_GAME_ACTION(ChangeMapSizeAction, GameCommand::ChangeMapSize, GameActions::Result)
+{
+private:
+    int32_t _sizeChange;
+
+public:
+    ChangeMapSizeAction() = default;
+    ChangeMapSizeAction(const int32_t sizeChange);
```

Comment:
I'm not a fan of this parameter, this should be enum class, there are only two possible inputs which is Increase, Decrease, passing -1 or 1 is just confusing, need to also consider that plugins may get access.

Reply:
For plugins you would probably just want to specify the end size you want to get to. In general that is how the rest of the game actions work (perhaps not all of them though).

Reply:
I also thought about just specifying the size instead, you can work out from current size if its decreasing or increasing but we can do that afterwards I guess.

Reply:
I like the idea of specifying the size. It is more future proof, seems like a simpler API for the action and might help resolve conflicts where two players change the map size at the same time.

Reply:
so what is the conclusion change the API as part of this PR?

Reply:
Well if @frutiemax is willing to refactor this PR I would be for the target size parameter instead of delta changes.